vita3k/patch/src/patch.cpp

```diff
@@ -0,0 +1,81 @@
+// Vita3K emulator project
+// Copyright (C) 2024 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include "patch/patch.h"
+
+#include <util/log.h>
+
+std::vector<Patch> get_patches(fs::path &path, const std::string &titleid) {
+    // Find a file in the path with the titleid
+    std::vector<Patch> patches;
+
+    LOG_INFO("Looking for patches for titleid {}", titleid);
+
+    for (auto &entry : fs::directory_iterator(path)) {
+        auto filename = entry.path().filename().string();
+        // Just in case users decide to use lowercase filenames
+        auto upper_filename = std::transform(filename.begin(), filename.end(), filename.begin(), ::toupper);
+
+        if (filename.find(titleid) != std::string::npos && filename.ends_with(".VPATCH")) {
```

Comment:
why `.VPATCH`?
Is this format used anywhere?
In VitaGrafix it is txt, so I think that would be better.

Reply:
I wanted it to be distinct because, while VitaGrafix uses `.txt`, this system isn't compatible with VitaGrafix patches, so it might be misleading (ie users may think they can just put the VitaGrafix `patchlist.txt` into the `patch` folder, which would not work).

If this isn't really a concern though, I am totally fine with just making them `.txt` 👍

Reply:
Changed to `.txt` since it probably isn't much of a concern